core_v2/sources/v2_domains.move

```diff
@@ -0,0 +1,1318 @@
+            } else {
+                false
+            }
+        } else {
+            false
+        };
+
+
+        // log the event
+        event::emit_event<RenewNameEvent>(
+            &mut borrow_global_mut<RenewNameEvents>(@aptos_names_v2).renew_name_events,
+            RenewNameEvent {
+                domain_name,
+                subdomain_name: option::none(),
+                renewal_fee_octas: price,
+                expiration_time_secs: record.expiration_time_sec,
+                target_address: record.target_address,
+                is_primary_name,
+            },
+        );
+    }
+
+    // === SUBDOMAIN MANAGEMENT ===
+
+    /// Disable or enable subdomain owner from transferring subdomain as domain owner
+    public fun set_subdomain_transferability_as_domain_owner(
+        router_signer: &signer,
+        sign: &signer,
+        domain_name: String,
+        subdomain_name: String,
+        transferrable: bool
+    ) acquires CollectionCapability, NameRecord, SubdomainExt {
+        assert!(address_of(router_signer) == @router_signer, error::permission_denied(ENOT_ROUTER));
+        validate_subdomain_registered_and_domain_owned_by_signer(sign, domain_name, subdomain_name);
+        let name_record_address = get_token_addr(domain_name, option::some(subdomain_name));
+        let transfer_ref = &borrow_global_mut<NameRecord>(name_record_address).transfer_ref;
+        if (transferrable) {
+            object::enable_ungated_transfer(transfer_ref);
+        } else {
+            object::disable_ungated_transfer(transfer_ref);
+        }
+    }
+
+    public fun transfer_subdomain_owner(
+        sign: &signer,
+        domain_name: String,
+        subdomain_name: String,
+        new_owner_address: address,
+        new_target_address: Option<address>,
+    ) acquires CollectionCapability, NameRecord, SubdomainExt, ReverseRecord, SetReverseLookupEvents {
+        // validate user own the domain
+        let signer_addr = signer::address_of(sign);
+        assert!(
+            is_token_owner(signer_addr, domain_name, option::none()),
+            error::permission_denied(ENOT_OWNER_OF_DOMAIN)
+        );
+        assert!(
+            !is_name_expired(domain_name, option::none()),
+            error::permission_denied(ECANNOT_TRANSFER_SUBDOMAIN_WHILE_DOMAIN_HAS_EXPIRED)
+        );
+
+        let token_addr = get_token_addr_inline(domain_name, option::some(subdomain_name));
+        let record = borrow_global_mut<NameRecord>(token_addr);
+        record.target_address = new_target_address;
+        object::transfer_with_ref(object::generate_linear_transfer_ref(&record.transfer_ref), new_owner_address);
+        // clear the primary name
+        clear_reverse_lookup_for_name(option::some(subdomain_name), domain_name);
+    }
+
+    /// this is for domain owner to update subdomain expiration time
+    public fun set_subdomain_expiration(
```

Comment:
This is a bit of strange experience. If I want to change my subdomain from auto renew to having a custom expiration, I would need to do two steps: 
1. Call set_subdomain_expiration_policy to change the policy
2. Call set_subdomain_expiration to change the expiration to custom
Seems a bit complex